Models/ProjectModels/Message/Catalog/MsgModelsFactory.php

```diff
@@ -0,0 +1,55 @@
+<?php
+declare(strict_types=1);
+
+namespace Models\ProjectModels\Message\Catalog;
+
+use Models\AbstractProjectModels\Message\Catalog\AbstractBaseMsgModel;
+use Models\ProjectModels\Message\Catalog\Show;
+//use mysql_xdevapi\Exception;
+
+class MsgModelsFactory
+{
+
+    public static function getMsgModel(string $customerType, string $actionType = null): AbstractBaseMsgModel
+    {
+        if (strtolower($customerType) === 'head_admin') {
+            switch (strtolower($actionType)) {
+                case 'show' :
+                    return new Show\Admin\Head\MsgModel();
+                default :
+                    throw new \Exception(
+                        "ActionType : '$actionType' in $customerType" . 'Controller' . " doesn't exist!"
```

Comment:
you have a lot of duplicates of showing case and throwing exception inside this module. Can you refactor it? 
For example : 

```
public static function getMsgModel(string $customerType, string $actionType = null): AbstractBaseMsgModel
{
    $customerType = strtolower($customerType);
    $actionType = strtolower($actionType);

    $namespace = '';

    switch ($customerType) {
        case 'head_admin':
            $namespace = 'Show\Admin\Head';
            break;
        case 'admin':
            $namespace = 'Show\Admin';
            break;
        case 'user':
            $namespace = 'Show\User';
            break;
        case 'not_logged':
            $namespace = 'Show';
            break;
        default:
            throw new \Exception("UserType: '$customerType' doesn't exist!");
    }

    switch ($actionType) {
        case 'show':
            $className = 'MsgModel';
            break;
        default:
            throw new \Exception("ActionType: '$actionType' in $customerType" . 'Controller' . " doesn't exist!");
    }

    $classNameWithNamespace = $namespace . '\\' . $className;
    
    if (!class_exists($classNameWithNamespace)) {
        throw new \Exception("Class '$classNameWithNamespace' doesn't exist!");
    }

    return new $classNameWithNamespace();
}
```

Or write additional private method where you thrown this exception with this text in order to have a less code in the method